activestorage/app/models/active_storage/blob/analyzable.rb

```diff
@@ -49,14 +49,16 @@ def analyzed?
 
   private
     def extract_metadata_via_analyzer
-      analyzer.metadata.merge(analyzed: true)
+      analyzers.reduce(ActiveSupport::HashWithIndifferentAccess.new) do |metadata, analyzer|
+        metadata.deep_merge!(analyzer.metadata)
+      end.merge!(analyzed: true)
```

Comment:
@p8 in this case, the metadata is only ever marked as `analyzed: true` once the array has been enumerated.